includes/Core/Authentication/Profile.php

```diff
@@ -115,28 +111,22 @@ public function has() {
 	 */
 	private function retrieve_google_profile_from_api() {
 
-		// Fall back to the user's WordPress profile information.
-		$current_user = wp_get_current_user();
-		$profile_data = array(
-			'email'     => $current_user->user_email,
-			'photo'     => get_avatar_url( $current_user->user_email ),
-			'timestamp' => 0, // Don't cache WP user data.
-		);
```

Comment:
Why was this code removed? If a user is logged in, we should still rely on this as default for user data, e.g. if the People API request fails for whatever reason.

We should probably make sure that in that case the value is cached for only a shorter time, e.g. 1 hour. Plus, we should only run this if `is_user_logged_in()`.

Reply:
> Why was this code removed? If a user is logged in, we should still rely on this as default for user data, e.g. if the People API request fails for whatever reason.

The main reason is that profile data should always be for the Google account that the user authenticated with and falling back to WP user data is not something I think we should mix in there unless it is only used for presentational purposes (although this could still be misleading). This came up in conversation previously as a way to get the email address of the authenticated Google account for permission checking, although I don't think it is currently used in that way. I think the chances of that request failing are likely due to authentication so the user would need to reauthenticate anyway. Also, if the request failed for some other reason, it would retry on the next admin request, so it's probably not a problem.

> Plus, we should only run this if is_user_logged_in()

I thought about that, except the state for which user is set in the `User_Options` which isn't available outside of it so we don't want to assume that the profile data is for the current user, which is partially why I removed the fallback data. If we want `Profile` to always represent the current user, we could remove it as a constructor dependency and take `Context` instead and then instantiate it internally to ensure that it is always for the current user.

Other solutions I considered would be adding a getter for the `user_id` on the user options class, or even re-implementing `Profile` as a sub-class of `User_Options` (not possible because it's `final`).

Also, it is currently only run if the user is logged in as it is only called during `\Google\Site_Kit\Core\Authentication\Authentication::inline_js_admin_data`.

Reply:
Agreed, it doesn't make sense to access the current user from `Profile`, while not being sure whether `$user_options` references the same user.

However, we could move the "fallback" logic to the JS function so that it is set by default. If `Profile::get()` is successful then, that data should override the existing user data coming from WP.